extensions/socket/internal.m

```diff
@@ -0,0 +1,657 @@
+#import <LuaSkin/LuaSkin.h>
+#import <CocoaAsyncSocket/GCDAsyncSocket.h>
+#import "socket.h"
+
+// Definitions
+@interface HSAsyncSocket : GCDAsyncSocket
+@property int readCallback;
+@property int writeCallback;
+@property int connectCallback;
+@property NSTimeInterval timeout;
+@property NSMutableArray* connectedSockets;
+@end
+
+// Userdata for hs.socket objects
+static const char *USERDATA_TAG = "hs.socket";
+#define getUserData(L, idx) (__bridge HSAsyncSocket *)((asyncSocketUserData *)lua_touserdata(L, idx))->asyncSocket;
+
+
+// Delegate implementation
+@implementation HSAsyncSocket
+
+- (id)init {
+    self.readCallback = LUA_NOREF;
+    self.writeCallback = LUA_NOREF;
+    self.connectCallback = LUA_NOREF;
+    self.timeout = -1;
+    self.connectedSockets = [[NSMutableArray alloc] init];
+    return [super initWithDelegate:self delegateQueue:dispatch_get_main_queue()];
+}
+
+- (void)socket:(HSAsyncSocket *)sock didConnectToHost:(NSString *)host port:(UInt16)port {
```

Comment:
The most reliable way to do this is to add class methods for the logX: methods... anything that relies on [LuaSkin shared] will be rejected before the method is ever entered. (e.g. `[LuaSkin logWarn:...]` rather than/in addition to `[[LuaSkin shared] logWarn:...]`

It's not a bad idea, maybe even include a prefix indicating when it's from another thread; @cmsj objections to my adding this?  I'd also like to see `hs.logger` used in the Lua side of the output so verbose and debug can be turned on and off as desired and can add that at the same time.